src/mscorlib/src/System/Reflection/MethodBase.Unsharable.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
I would call it MethodBase.CoreCLR.cs - same naming convention is used for CoreCLR specific partial classes in corefx.